src/IR/DropInference.hs

```diff
@@ -168,7 +172,7 @@ insertDropAlt (I.AltDefault var, exprReturn) exprMatch = do
   return (I.AltDefault var, retExpr')
 
 -- | Inserting dup/drops into AltData arms
--- | TODO: type is currently wrong and is only a placeholder
+-- | TODO: type info is currently unavailable and is only a placeholder
```

Comment:
Can you explain more about this TODO?

Reply:
This TODO is addressing the problem that I do not have the types of DCon's fields from upstream.  
For example if a pattern is
`match x`
`MyData a b -> ...`
the types of `a` and `b` are unavailable because the names `a` and `b` are given in a list of VarIds without other information about them.  The current treatment is to use a placeholder for their types, which I believe to cause no problem because it only affects the types of dup/drop expressions (checked during runtime anyways).